busybox-udhcpd: Move wireless udhcpd configuration file from os-common

gratian opened the pull request:
Signed-off-by: Gratian Crisan <gratian.crisan@ni.com>

gratian committed:
busybox-udhcpd: Move wireless udhcpd configuration file from os-common

Signed-off-by: Gratian Crisan <gratian.crisan@ni.com>

gratian commented:
Attn: @ni/rtos

chaitu236 commented on recipes-core/busybox/busybox_1.31.%.bbappend:
```diff
@@ -27,6 +28,7 @@ DEPENDS =+ " libselinux"
 FILES_${PN}-ifplugd = "${sysconfdir}/init.d/busybox-ifplugd ${sysconfdir}/ifplugd/ifplugd.action ${sysconfdir}/ifplugd/ifplugd.conf"
 FILES_${PN}-acpid = "${sysconfdir}/init.d/busybox-acpid ${sysconfdir}/acpid.conf ${sysconfdir}/acpi ${sysconfdir}/acpi/poweroff.sh"
 FILES_${PN}-zcip = "${sysconfdir}/natinst/networking/zcip.script"
+FILES_${PN}-udhcpd =+ "${sysconfdir}/udhcpd.wlan0.conf"
```
Should busybox-udhcpd be added to packagegroup-ni-base.bb?

gratian replied:
I did not add the busybox-udhcpd package (I'm just adding a file to it) so I was assuming is already in the image. Not sure what pulls it in. I'll see if I can track down the dependency chain.

gratian replied:
Btw. busybox-udhcpc is in the same boat i.e. AFAICT not listed explicitly in packagegroup-ni-base.bb but used in meta-nilrt.

chaitu236 replied:
There is a busybox-udhcpd in feeds (which only has an initscript) but its not in the image. The udhcpd binary itself is part of busybox package. So we have to explicitly add it to packagegroup for this conf file to get installed. But I don't think we'd want to install the initscript that comes with the package as we run udhcp ourself.

chaitu236 replied:
It looks like busybox-udhcpc ends up getting installed in base image as busybox has recommends dependency on it, but thats not the case with busybox-udhcpd

gratian replied:
> So we have to explicitly add it to packagegroup for this conf file to get installed. But I don't think we'd want to install the initscript that comes with the package as we run udhcp ourself.

Well that's unfortunate. I guess I'll have to figure out how to remove the init script in our bbappend.

chaitu236 replied:
Adding following seems to atleast prevent creation of postinst (that runs update-rc.d).

```
unset INITSCRIPT_NAME_${PN}-udhcpd
INITSCRIPT_PACKAGES_remove = "${PN}-udhcpd"
```

Don't know how to prevent the file from getting installed though, but maybe this is sufficient?

gratian replied:
@chaitu236 nice catch and thanks for the suggestions. Rebased on latest nilrt/master/dunfell and pushed a new commit that adds busybox-udhcpd to packagegroup-ni-base.bb with the modifications discussed here.

jeminor commented:
Looks good to me after chaitu236's suggestion.

gratian committed:
packagegroup-ni-base, busybox: Add dependency on busybox-udhcpd

Commit 3221e817ebc2 ("busybox-udhcpd: Move wireless udhcpd
configuration file from os-common") adds a configuration file to the
busybox-udhcpd package.

This configuration file has to be included in the base NILRT image.
Add it to the corresponding package group.

Additionally the default busybox-udhcpd recipe installs an init script
which is not used in NILRT images. On these images udhcpd is invoked
directly from ifplugd action scripts. Remove it.

Suggested-by: Chaitanya Vadrevu <chaitanya.vadrevu@ni.com>
Signed-off-by: Gratian Crisan <gratian.crisan@ni.com>
